orderbook/src/bad_token_cache.rs

```diff
@@ -0,0 +1,53 @@
+use crate::bad_token::{BadTokenDetector, TokenQuality};
+use anyhow::Result;
+use primitive_types::H160;
+use std::{collections::HashMap, sync::Mutex};
+
+pub struct BadTokenCache {
+    detector: BadTokenDetector,
+    // Std mutex is fine because we don't hold lock across await.
+    cache: Mutex<HashMap<H160, TokenQuality>>,
+    // Explicitly allowed tokens that are always treated as good.
+    allow_list: Vec<H160>,
+}
+
+impl BadTokenCache {
+    pub fn new(detector: BadTokenDetector, allow_list: Vec<H160>) -> Self {
+        Self {
+            detector,
+            cache: Default::default(),
+            allow_list,
+        }
+    }
+
+    pub async fn is_good(&self, token: H160) -> Result<bool> {
```

Comment:
I decided to keep this a simple bool for now until I know we need the fool quality enum.

Reply:
I think that's totally fine. Its super easy to make it return a `TokenQuality` if its needed in the future.